core/templates/pages/contributor-dashboard-page/modal-templates/question-suggestion-editor-modal.controller.ts

```diff
@@ -80,7 +80,32 @@ angular.module('oppia').controller('QuestionSuggestionEditorModalController', [
       return QuestionValidationService.isQuestionValid(
         $scope.question, $scope.misconceptionsBySkill);
     };
-
+    $scope.skillId = $scope.skill.getId();
+    $scope.onClickChangeDifficulty = function() {
+      $uibModal.open({
+        templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+          '/pages/topic-editor-page/modal-templates/' +
+          'select-skill-and-difficulty-modal.template.html'),
+        backdrop: true,
+        resolve: {
+          skillId: () => $scope.skillId
+        },
+        controller: (
+          'QuestionsOpportunitiesSelectSkillAndDifficultyModalController')
+      }).result.then(function(result) {
+        if (AlertsService.warnings.length === 0) {
+          $scope.skillDifficulty = result.skillDifficulty;
+          skillDifficulty = result.skillDifficulty;
+          $scope.skillDifficultyString = Object.entries(
+            SKILL_DIFFICULTY_LABEL_TO_FLOAT).find(
+            entry => entry[1] === skillDifficulty)[0];
+        }
```

Comment:
Can we extract this into a separate function which we can use on line 54?

Reply:
done. In PR #12403